common/solver/batch_bicgstab_kernels.hpp.inc

```diff
@@ -285,127 +178,67 @@ __device__ __forceinline__ void update_x_middle(
 
 template <typename StopType, typename PrecType, typename LogType,
           typename BatchMatrixType, typename ValueType>
-__global__ void apply_kernel(
-    const int max_iter, const gko::remove_complex<ValueType> tol,
-    LogType logger, PrecType prec_shared, const BatchMatrixType a,
-    const gko::batch_dense::UniformBatch<const ValueType> left,
-    const gko::batch_dense::UniformBatch<const ValueType> right,
-    const gko::batch_dense::UniformBatch<ValueType> b,
-    const gko::batch_dense::UniformBatch<ValueType> x)
+__global__ void apply_kernel(const int shared_gap, const int max_iter,
```

Comment:
Is there a particular reason to pass this shared_gap to the function ? And not just infer it from num_rows ?

Reply:
It is indeed inferred from num_rows, but I don't want to repeat that calculation logic, so I'm passing it to the kernel. It turns out to be slightly more performant with the shared memory allocation rounded to 32.